src/WorkspaceRegistration/WorkspaceRegistration.tsx

```diff
@@ -0,0 +1,299 @@
+          } else {
+            // eslint-disable-next-line no-console
+            console.error('Requestor returns 201 but no request_id in payload body'); // shouldn't get here
+          }
+        } else if (status === 409) {
+          // there is already a request for this user on this study, display a message and disable the button
+          setFormSubmissionButtonDisabled(true);
+          setFormSubmissionStatus({ status: 'warning', text: 'There is already a pending request for workspace access, please wait while we are processing your request.' });
+        } else {
+          // something has gone wrong
+          setFormSubmissionStatus({ status: 'error', text: `Failed to create a request with error code: ${status}. Please try again later. If the error persists, please contact us for help.` });
+        }
+        setReqAccessRequestPending(false);
+      },
+    );
+  };
+
+  const onFinish = (values) => {
+    setReqAccessRequestPending(true);
+    handleRegisterFormSubmission(values);
+  };
+
+  const onReset = () => {
+    form.resetFields();
+  };
+
+  const onRadioChange = (e) => {
+    setRole(e.target.value);
+  };
+
+  if (formSubmissionStatus) {
+    return (
+      <div className='workspace-reg-container'>
+        <div className='workspace-reg-form-container'>
+          {(formSubmissionStatus.status === 'success') ? (
+            <Result
+              status={formSubmissionStatus.status}
+              title='Your access request has been submitted!'
+              subTitle='Thank you for your submission. Requests take up to 1 business day to complete. You will be notified of the status.'
+              extra={[
+                <Link key='discovery' to={'/discovery'}>
+                  <Button>Go To Discovery Page</Button>
+                </Link>,
+              ]}
+            />
+          ) : (
+            <Result
+              status={formSubmissionStatus.status}
+              title='A problem has occurred during submitting the request!'
+              subTitle={formSubmissionStatus.text}
+              extra={[
+                <Button type='primary' key='close' onClick={() => { setFormSubmissionStatus(null); setReqAccessRequestPending(false); }}>
+                  Close
+                </Button>,
+              ]}
+            />
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  if (kayakoConfig) {
+    return (
+      <div className='workspace-reg-container'>
+        <div className='workspace-reg-form-container'>
+          <Form className='workspace-reg-form' {...layout} form={form} name='workspace-reg-request-form' onFinish={onFinish} validateMessages={validateMessages}>
+            <Divider plain>Workspace Registration Access Request</Divider>
+            <Typography style={{ textAlign: 'center' }}>
+              Please fill out this form to request and be approved for access to workspace with the BRH Platform.
```

Comment:
Don't hard code BRH in here. All hard coded texts should be generic. If you have to mention a specific env name, make it configurable

Reply:
That is a very good point. I always am an advocate against hardcoding, thank you for catching this.